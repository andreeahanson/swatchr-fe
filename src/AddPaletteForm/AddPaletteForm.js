import React, { Component } from "react";
import "./AddPaletteForm.scss";

class AddPaletteForm extends Component {
  state = {
    name: "",
    displayInput: false
  };

  toggleInput = () => {
    this.setState({ displayInput: !this.state.displayInput });
  };

  handleChange = e => {
    this.setState({ name: e.target.value });
  };

  handleSubmit = e => {
    if (e.keyCode === 13) {
      e.preventDefault();
      const id = this.props.currentProjectId;
      const newPalette = {
        palette: {
          name: this.state.name,
          project_id: id,
          color1: this.props.colors[0].hex,
          color2: this.props.colors[1].hex,
          color3: this.props.colors[2].hex,
          color4: this.props.colors[3].hex,
          color5: this.props.colors[4].hex
        }
      };
      this.props.postFetchPalette(newPalette, id);
      this.clearInput();
    }
  };

  clearInput = () => {
    this.setState({
      name: ""
    });
  };

  render() {
    return (
      <div className="sidebar-add-container">
        <img
          onClick={this.toggleInput}
          src="./save.png"
          alt="save icon"
        />
        {!this.state.displayInput && <p>Save palette</p>}
        {this.state.displayInput && (
          <input
            type="text"
            value={this.state.name}
            onChange={this.handleChange}
            className="input-animation"
            onKeyDown={this.handleSubmit}
          />
        )}
      </div>
    );
  }
}

export default AddPaletteForm;
