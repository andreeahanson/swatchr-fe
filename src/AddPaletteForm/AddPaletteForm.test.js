import React from 'react';
import { shallow } from 'enzyme';
import AddPaletteForm from './AddPaletteForm';

describe('AddPaletteForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddPaletteForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when user types in input', () => {
    const mockEvent =  { target: { name: "name", value: "test name" } }
    const expected = "test name"
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should update state when toggleInput is called', () => {
    wrapper.instance().toggleInput()
    expect(wrapper.state('displayInput')).toEqual(true);
  })
})