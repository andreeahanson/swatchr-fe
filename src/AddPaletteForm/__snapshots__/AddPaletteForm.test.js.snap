// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPaletteForm should match the snapshot 1`] = `ShallowWrapper {}`;
